### PR TITLE
add danger-ruby to CI

### PR DESCRIPTION
**Pull request by omgitsbillryan**
## Description of change
Connects: https://github.com/department-of-veterans-affairs/va.gov-team/issues/1934

- Adds [ruby-danger](https://danger.systems/ruby/) to our CI
- Adds 2 danger rules:
  - warn if PR is too big
  - warn if both `app/` & `db/` files are modified in the same PR

image

(Picture source: https://github.com/department-of-veterans-affairs/vets-api/pull/3332#issuecomment-538814142)

## Testing done
Testing was done in a separate WiP PR: https://github.com/department-of-veterans-affairs/vets-api/pull/3332

## Testing planned
None.

#### Applies to all PRs

- [x] Appropriate logging
- [x] Swagger docs have been updated, if applicable
- [x] Provide link to originating GitHub issue, or connected to it via ZenHub
- [x] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [x] Provide which alerts would indicate a problem with this functionality (if applicable)

### CODE REVIEW

File: docker-compose.test.yml [hunk 1]
```diff
@@ -31,6 +31,10 @@ services:
       CI:
       RAILS_ENV: test
       CC_TEST_REPORTER_ID: '0c396adc254b0317e2c3a89a1c929fd61270b133c944d3e9c0f13b3937a7ce45'
+      CHANGE_ID:           "${CHANGE_ID}"
+      GIT_URL:             "${GIT_URL}"
+      JENKINS_URL:         "${JENKINS_URL}"
+      DANGER_GITHUB_API_TOKEN: "${DANGER_GITHUB_API_TOKEN}"
```
Comment (omgitsbillryan) [inline docker-compose.test.yml]: These are all required by `danger` to function properly.

File: really_big_md.md [hunk 2]
```diff
@@ -0,0 +1,292 @@
+### Big markdown to trigger ruby-danger PR size warning
```
Comment (omgitsbillryan) [inline really_big_md.md]: Pls ignore this file. Just meant to trigger the danger-ruby PR size rule.

**Comment by omgitsbillryan** [comment]
Looks like some lingering `.git` files owned by `root` were left in the jenkins workspace. I can clean them up for now, but I should make sure that this won't happen again.

File: Makefile [hunk 3]
```diff
@@ -19,7 +19,7 @@ ci:
 .PHONY: clean
 clean:
 	rm -r data || true
-	$(COMPOSE_TEST) run vets-api rm -r coverage log tmp || true
+	$(COMPOSE_TEST) run vets-api rm -r coverage log tmp .git || true
```
Comment (omgitsbillryan) [inline Makefile]: A stale `.git/` owned by `root` was being left behind causing the Jenkinsfile `post` step to fail at `deleteDir()`. I'm guessing that `danger` probably does some git-stuff under the hood and leaves this stale relic behind.

File: Dangerfile [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# PR too big
+MAX_PR_SIZE = 250
+warn("PR is exceeds `#{MAX_PR_SIZE}` LoC. Consider breaking up into multiple smaller ones.") if git.lines_of_code > MAX_PR_SIZE
+
+
```
Comment (kfrz) [inline Dangerfile]:
```suggestion
```

Remove extra whitespace

File: app/controllers/application_controller.rb [hunk 5]
```diff
@@ -7,6 +7,7 @@
 require 'sentry_logging'
 require 'aes_256_cbc_encryptor'
 
+# random change to app/
```
Comment (kfrz) [inline app/controllers/application_controller.rb]: Will these random changes will be removed before merging the PR?

File: Dangerfile [hunk 6]
```diff
@@ -0,0 +1,22 @@
+# PR too big
+MAX_PR_SIZE = 250
+warn("PR is exceeds `#{MAX_PR_SIZE}` LoC. Consider breaking up into multiple smaller ones.") if git.lines_of_code > MAX_PR_SIZE
```
Comment (kfrz) [inline Dangerfile]:
```suggestion
if git.lines_of_code > MAX_PR_SIZE
  warn("Pull request exceeds `#{MAX_PR_SIZE}` lines of code. Consider breaking up into multiple smaller pull requests.")
end
```

Nitpicky but a bit easier to parse the `if` block imo

File: Dangerfile [hunk 7]
```diff
@@ -0,0 +1,22 @@
+# PR too big
```
Comment (kfrz) [inline Dangerfile]:
```suggestion
# Warn if a pull request is too big
```

File: Dangerfile [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+MAX_PR_SIZE = 250
+warn("PR is exceeds `#{MAX_PR_SIZE}` LoC. Consider breaking up into multiple smaller ones.") if git.lines_of_code > MAX_PR_SIZE
+
+
+# simultaneous migration & app code warning
```
Comment (kfrz) [inline Dangerfile]:
```suggestion
# Warn when a PR includes a simultaneous DB migration and application code changes
```

File: app/controllers/application_controller.rb (hunk 5 above)
Comment (omgitsbillryan) [reply app/controllers/application_controller.rb]: Yep

File: Dangerfile (hunk 6 above)
Comment (omgitsbillryan) [reply Dangerfile]: Agree.

**Comment by omgitsbillryan** [comment]
@kfrz 
> Does danger need to be in the :development and :test group? 

Yes, because danger can be executed locally - which is much faster than trying to make a change and wait for CI to run.

```
bundle exec danger pr https://github.com/department-of-veterans-affairs/vets-api/pull/3431
```

image

You can even step into your `Dangerfile` changes with `pry` 
```
bundle exec danger local --pry
```

> Does the CI server use RAILS_ENV=test or production

[It uses RAILS_ENV=test](https://github.com/department-of-veterans-affairs/vets-api/blob/master/Jenkinsfile#L32)

**Pull request merged**